phoenix-core/src/main/java/org/apache/phoenix/coprocessor/TaskRegionObserver.java

```diff
@@ -0,0 +1,261 @@
+import org.apache.hadoop.hbase.coprocessor.RegionCoprocessorEnvironment;
+import org.apache.hadoop.hbase.coprocessor.RegionObserver;
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+import org.apache.phoenix.jdbc.PhoenixConnection;
+import org.apache.phoenix.jdbc.PhoenixDatabaseMetaData;
+import org.apache.phoenix.jdbc.PhoenixDriver;
+import org.apache.phoenix.query.QueryServices;
+import org.apache.phoenix.query.QueryServicesOptions;
+import org.apache.phoenix.schema.MetaDataClient;
+import org.apache.phoenix.schema.PTable;
+import org.apache.phoenix.schema.PTable.TaskType;
+
+import org.apache.phoenix.util.PhoenixRuntime;
+import org.apache.phoenix.util.QueryUtil;
+
+
+/**
+ * Coprocessor for task related operations. This coprocessor would only be registered
+ * to SYSTEM.TASK table
+ */
+
+public class TaskRegionObserver implements RegionObserver, RegionCoprocessor {
+    public static final Log LOG = LogFactory.getLog(TaskRegionObserver.class);
+    protected ScheduledThreadPoolExecutor executor = new ScheduledThreadPoolExecutor(TaskType.values().length);
+    private long timeInterval = QueryServicesOptions.DEFAULT_TASK_HANDLING_INTERVAL;
+    private long timeMaxInterval = QueryServicesOptions.DEFAULT_TASK_HANDLING_MAX_INTERVAL;
+    @GuardedBy("TaskRegionObserver.class")
+    // Added for test purposes
+    private long initialDelay = 0;
+
+    @Override
+    public void preClose(final ObserverContext<RegionCoprocessorEnvironment> c,
+            boolean abortRequested) {
+        executor.shutdownNow();
+    }
+
+    @Override
+    public Optional<RegionObserver> getRegionObserver() {
+        return Optional.of(this);
+    }
+
+    @Override
+    public void start(CoprocessorEnvironment env) throws IOException {
+        Configuration config = env.getConfiguration();
+        timeInterval =
+                config.getLong(
+                    QueryServices.TASK_HANDLING_INTERVAL_ATTRIB,
+                    QueryServicesOptions.DEFAULT_TASK_HANDLING_INTERVAL);
+        timeMaxInterval =
+                config.getLong(
+                        QueryServices.TASK_HANDLING_MAX_INTERVAL_ATTRIB,
+                        QueryServicesOptions.DEFAULT_TASK_HANDLING_MAX_INTERVAL);
+        initialDelay =
+                config.getLong(
+                    QueryServices.TASK_HANDLING_INITIAL_DELAY_ATTRIB,
+                    QueryServicesOptions.DEFAULT_TASK_HANDLING_INITIAL_DELAY);
+    }
+    
+    @Override
+    public void postOpen(ObserverContext<RegionCoprocessorEnvironment> e) {
+        final RegionCoprocessorEnvironment env = e.getEnvironment();
+
+        // turn off verbose deprecation logging
+        Logger deprecationLogger = Logger.getLogger("org.apache.hadoop.conf.Configuration.deprecation");
+        if (deprecationLogger != null) {
+            deprecationLogger.setLevel(Level.WARN);
+        }
+        try {
+            Class.forName(PhoenixDriver.class.getName());
```

Comment:
We dont need this thing anymore since Java 1.5+, should be okay to remove.

Reply:
Will do